### PR TITLE
Logging in framework.cc should use the session logger

### PR DESCRIPTION
**Pull request by yufenglee**
**Description**: Describe your changes.
Use session logger for logging in framework.cc
**Motivation and Context**
default logger action can not be controller by the log level option in SessionOption
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/core/framework/execution_frame.cc [hunk 1]
```diff
@@ -337,7 +337,7 @@ Status ExecutionFrame::AllocateMLValueTensorPreAllocateBuffer(OrtValue& ort_valu
 
     // be generous and use the buffer if it's large enough. log a warning though as it indicates a bad model
```
Comment (pranavsharma) [inline onnxruntime/core/framework/execution_frame.cc]: @yufenglee the comment says that could be a bad model. Can you look into that as well?

**Comment by pranavsharma** [comment]
You'll need 2 reviewers to sign-off since this is for the release branch.

File: onnxruntime/core/framework/execution_frame.cc (hunk 1 above)
Comment (yufenglee) [reply onnxruntime/core/framework/execution_frame.cc]: The root cause is that shape inference for Slice with dim_param is not correct, but ORT also has this logging issue. This is the fix of ORT side.

**Pull request merged**